cmd/manager/main.go

```diff
@@ -115,6 +118,20 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Upgrade the components if required
+	k8sClient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	log.Info("Performing upgrade operation")
```

Comment:
The message can be something like `Check if any of the CRs have to be upgraded`.